packages/terra-date-picker/src/DateUtil.js

```diff
@@ -508,7 +515,14 @@ class DateUtil {
    * @return {moment} The decremented moment date
    */
   static decrementDateByDay(date, format) {
-    return moment(date, format, true).subtract(1, 'd').format('YYYY-MM-DD');
+    const momentDate = moment(date, format, true).subtract(1, 'd');
+    if (momentDate.year() < Number(DateUtil.MIN_YEAR)) {
+      return DateUtil.MIN_DATE;
+    } else if (momentDate.year() > Number(DateUtil.MAX_YEAR)) {
+      return DateUtil.MAX_DATE;
```

Comment:
Same as the above comment where it should ever get in this scenario because the year should never be greater than 2100 to being with.

Reply:
We shouldn't get into the scenario right now, but I would prefer leaving it in for protecting future uses of method just in case.